### PR TITLE
[FIX] website: address test tour issue with links href updating

### PR DESCRIPTION
**Pull request by SergeBayet**
This commit resolves an issue in the `link_tools` test, specifically addressing a failure that arises when verifying the synchronization between the link's href and the content of the `<a>` tag. The failure was attributed to the oversight of not clicking outside the popup to close it before interacting with the link.

runbot-24929

### CODE REVIEW

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/static/tests/tours/link_tools.js [hunk 1]
```diff
@@ -226,14 +227,12 @@ wTourUtils.registerWebsitePreviewTour('link_tools', {
     {
         content: "Edit link label",
         trigger: "iframe .s_text_image p a",
-        run() {
+        run(actions) {
             // Simulating text input.
             const link = this.$anchor[0];
-            link.textContent = "callmemaybe.com/shop";
+            actions.text("callmemaybe.com/shops");
```
[the remark this answers is not in the document]
Comment (SergeBayet) [reply addons/website/static/tests/tours/link_tools.js:233]: I don't konw neither for the previous code. The new one is borrowed from this commit in [16.4](https://github.com/odoo-dev/odoo/blob/7d26c7b666033a16acfb430b6bf06235dd7d88ea/addons/website/static/tests/tours/link_tools.js), that I found easier to understand. I'll add some explanations to this step.

File: addons/website/static/tests/tours/link_tools.js [hunk 2]
```diff
@@ -213,6 +213,7 @@ wTourUtils.registerWebsitePreviewTour('link_tools', {
         trigger: "#o_link_dialog_url_input",
         run: "text callmemaybe.com",
     },
+    clickOnImgStep,
```
[the remark this answers is not in the document]
Comment (SergeBayet) [reply addons/website/static/tests/tours/link_tools.js:216]: I didn't know about `text_blur`. I'm gonna try... thanks

**Comment by SergeBayet** [comment]
`text_blur` doesn't trigger the event that closes the popover-tooltip. So I had to leave the 'clickOnImgStep' when required. Do you have an alternative solution?

**Review by qsm-odoo — APPROVED** [review]
- Retargeted to 16.0, despite the race condition being only in >= 16.3. In 16.0, the tour seems robust now but since the code is the same... I figured, better unify the tour definition given how annoying it is (breaking differently in each sub version).
- These changes do not apparently (over ~20 tests) change the amount of race condition... but they can only help. `text_blur` is what should have been used, etc etc.
- Kept the "input" dispatch that was initially removed by this PR: without it we do not go through `_onInput` which was the point of that step apparently.

After this PR (and its forward-port), the tour:
- Succeeds 100% of the time in <= 16.2
- Fails ~50% (difficult to say) of the time in >= 16.3 (still have to understand why and fix later).

@robodoo r+

**Comment by qsm-odoo** [comment]
Note for tracking: part of this PR was actually a backport of https://github.com/odoo/odoo/commit/7d26c7b666033a16acfb430b6bf06235dd7d88ea and I did not understand it until https://github.com/odoo/odoo/pull/149854#pullrequestreview-1828403500
More PR to follow... damned tour.

**Pull request closed without merging**